src/Microsoft.DotNet.ImageBuilder/src/Commands/CleanAcrImagesCommand.cs

```diff
@@ -233,5 +248,24 @@ private async Task DeleteRepositoryAsync(IContainerRegistryClient acrClient, Lis
         }
 
         private static bool IsExpired(DateTimeOffset dateTime, int expirationDays) => dateTime.AddDays(expirationDays) < DateTimeOffset.Now;
+
+        private async Task<bool> IsAnnotationManifestAsync(ArtifactManifestProperties manifest, IContainerRegistryContentClient acrContentClient)
+        {
+            ManifestQueryResult manifestResult = acrContentClient.GetManifestAsync(manifest.Digest).Result;
+
+            // An annotation is just a referrer and referrers are indicated by the presence of a subject field.
+            return manifestResult.Manifest["subject"] is not null;
+        }
+
+        private async Task<bool> HasExpiredEolAsync(ArtifactManifestProperties manifest, int expirationDays)
```

Comment:
Now this method doesn't need to be async either.

```suggestion
        private bool HasExpiredEol(ArtifactManifestProperties manifest, int expirationDays)
```